hawkbit: fix memory issue with mcuboot_img_header size

joelguittet opened the pull request:
This pull request fix memory issue with mcuboot_img_header size reported at https://github.com/zephyrproject-rtos/zephyr/issues/54459

joelguittet committed:
hawkbit: fix memory issue with mcuboot_img_header size

Fix a memory issue when calling boot_read_bank_header the length should be
the size of `struct mcuboot_img_header`.
Fixes #54459.

Signed-off-by: Joel Guittet <joelguittet@gmail.com>

de-nordic reviewed (dismissed):
Looks OK.

ycsin reviewed (dismissed):
looks good, thanks for this fix!

ycsin commented:
cc @nandojve
I saw a similar implementation in `updatehub_firmware.c`

ycsin commented on subsys/mgmt/hawkbit/hawkbit_firmware.c:
```diff
@@ -13,7 +13,7 @@ bool hawkbit_get_firmware_version(char *version, int version_len)
 
 	if (boot_read_bank_header(FIXED_PARTITION_ID(slot0_partition), &header,
-				  version_len) != 0) {
+				  sizeof(struct mcuboot_img_header)) != 0) {
```
Probably its better to replace this with `sizeof(header)` instead, so it simply follows the type of the `header` variable and more future proof

de-nordic replied:
Yeah, this is actually good point.

nandojve replied:
Is there any reason to not check the compatibility of header version?

https://github.com/zephyrproject-rtos/zephyr/blob/d4c8de514f8b223e8b76c16dbfad482a2ca92edf/subsys/mgmt/updatehub/updatehub_firmware.c#L27-L30

joelguittet replied:
`sizeof(header)` is fine for me too.
About the check of the version, this can be added, sure (not the original purpose of the PR but can be done).
Do you want 1 commit for the fix or 2 commits in the history ?

joelguittet replied:
@nandojve done!

joelguittet committed:
hawkbit: check mcuboot header version

Checking mcuboot header version prior to use the content of the header
structure.

Signed-off-by: Joel Guittet <joelguittet@gmail.com>
